proj2/src/wavcb.cpp

```diff
@@ -36,74 +36,53 @@ int main(int argc, char *argv[]) {
     }
     
     // initialize centroids
-    int index, cur_centroidvec_size;
     vector<vector<short>> centroids;
+    {set<int> indexesUsed;
+    int index, prev_centroidvec_size;
     for(int i=0; i<numCentroids; i++) {
-        cur_centroidvec_size = centroids.size();
-        while(centroids.size() == cur_centroidvec_size) {
+        prev_centroidvec_size = centroids.size();
+        while(centroids.size() == prev_centroidvec_size) {
             index = rand() % blocks.size();
-            if(!count(centroids.begin(),centroids.end(),blocks.at(index))) { // if centroids does not yet contain that possible centroid
-                centroids.push_back(blocks.at(index));
+            if (indexesUsed.count(index) == 0) {
```

Comment:
Since I'm doing a copy of the vectors, I can no longer compare the references of the vectors. To solve this I created a et` that holds the indexes of the blocks already assigned to centroids.